### PR TITLE
DO NOT MERGE: Added spec for flow log resource to networkWatcher.json

### PR DESCRIPTION
**Pull request by irrogozh**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [ ] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Review by anton-evseev — CHANGES_REQUESTED** [review]
Since this isn't a top-level resource, I would say it makes sense to use `NetworkWatcherFlowLog` instead of just `FlowLog`

**Review by KrisBash — CHANGES_REQUESTED** [review]
Please see comments.

File: specification/network/resource-manager/Microsoft.Network/stable/2019-07-01/networkWatcher.json [hunk 1]
```diff
@@ -2225,6 +2468,79 @@
...
+    "FlowLogPropertiesFormat": {
+      "description": "Parameters that define the configuration of flow log.",
+      "required": [
+        "targetResourceId",
+        "storageId",
+        "enabled"
+      ],
+      "properties": {
+        "targetResourceId": {
+          "description": "ID of network security group to which flow log will be applied.",
```
[the remark this answers is not in the document]
Comment (irrogozh) [reply specification/network/resource-manager/Microsoft.Network/stable/2019-07-01/networkWatcher.json]: it's in ARM manifest:
        {
          "actionName": "Microsoft.Network/networkWatchers/flowLogs/write",
          "linkedProperty": "properties.storageId",
          "linkedAction": "Microsoft.Storage/storageAccounts/listKeys/action"
        },

File: specification/network/resource-manager/Microsoft.Network/stable/2019-07-01/networkWatcher.json [hunk 2]
```diff
@@ -2225,6 +2468,79 @@
...
+          "type": "string"
+        },
+        "targetResourceGuid": {
+          "description": "GUID of network security group to which flow log will be applied.",
+          "type": "string"
+        },
+        "storageId": {
+          "description": "ID of the storage account which is used to store the flow log.",
```
[the remark this answers is not in the document]
Comment (irrogozh) [reply specification/network/resource-manager/Microsoft.Network/stable/2019-07-01/networkWatcher.json]: Updated in ARM manifest:
        {
          "actionName": "Microsoft.Network/networkWatchers/flowLogs/write",
          "linkedProperty": "properties.storageId",
          "linkedAction": "Microsoft.Storage/storageAccounts/listKeys/action"
        },

File: specification/network/resource-manager/Microsoft.Network/stable/2019-07-01/networkWatcher.json [hunk 3]
```diff
@@ -2225,6 +2468,79 @@
...
+          "description": "Properties of the flow log."
+        },
+        "etag": {
+          "type": "string",
+          "description": "A unique read-only string that changes whenever the resource is updated."
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "./network.json#/definitions/Resource"
```
[the remark this answers is not in the document]
Comment (irrogozh) [reply specification/network/resource-manager/Microsoft.Network/stable/2019-07-01/networkWatcher.json]: We already had proxy child resource. Now we need to make it tracked resource

**Pull request closed without merging**